CppCoreGuidelines.md

```diff
@@ -9548,7 +9530,8 @@ This is more efficient:
 
 ##### Reason
 
-If you first make an object and then give it to a `shared_ptr` constructor, you (most likely) do one more allocation (and later deallocation) than if you use `make_shared()` because the reference counts must be allocated separately from the object.
+`make_shared` gives a more concise statement of the construction.
+It also gives an opportunity to eliminate a separate allocation for the reference counts, by placing the `shared_ptr`'s use counts next to its object.
```

Comment:
Choosing the C.151 wording over R.22 wording for Reason because it's less wordy while mentioning more.